### DISCUSSION TITLE
Routes starting with a dot

### DISCUSSION THREAD

**Discussion by vekunz** [sveltejs/kit#7562] [General]
Hi, basically, file-based routing is a great concept for defining routes. But currently, we have a case where file-based routing does not work very well for us. Because of some OpenId Connect stuff, we need a route where one path segment has the name `.well-known`. Basically, this works well with SvelteKit, but this makes some problems with other tools. For example, Vitest, ESLint, and tsc (for type checking) ignore everything inside this folder. For Vitest, we worked around the problem, but generally speaking, it is not a good solution to work around any problem that a folder with a dot as the first character makes. It would be great if SvelteKit could implement a way that we can define a route such as  `.well-known`, but without writing the dot as the first character into the folder name.

**Comment by dominikg** [comment]
you could use a mandatory path param `routes/[wellknown=wellknown]` with the wellknown matcher only matching .well-known

or configure the other tools not ignore that directory

**Reply by vekunz** [reply]
Thank you, with a matcher, everything seems to work fine now.

**Comment by gterras** [comment]
Does this route really need to exist? You could detect the routeId and return a custom Response from handle hook.

**Reply by vekunz** [reply]
Yes, this route really needs to exist. We need it because of some openid connect jwks verification stuff.

**Reply by gterras** [reply]
I meant physically exist, you can do literally anything (including any kind of identity or ownership check) with a custom response and the requester wouldn't see the difference (IMO it's even cleaner to handle this strictly separately from normal routing).

**Reply by vekunz** [reply]
I do not think, that it would be cleaner to handle this in an handle hook. This is not some basic auth stuff or so, but the routes return information that other services depends on. This has no place in a hook. This is stuff for a very normal +server.ts endpoint.

**Comment by dummdidumm** [comment]
The recommended way to do this now is to use encodings to represent the dot. See the info box at the bottom of [this section](https://kit.svelte.dev/docs/advanced-routing#encoding) for more details.
